src/record_syscall.cc

```diff
@@ -3793,7 +3793,7 @@ static void process_mremap(RecordTask* t, remote_ptr<void> old_addr,
 
   if (t->trace_writer().write_mapped_region(t, km, st) ==
       TraceWriter::RECORD_IN_TRACE) {
-    off64_t end = (off64_t)st.st_size - km.file_offset_bytes();
+    off64_t end = max((off64_t)st.st_size - km.file_offset_bytes(), 0UL);
```

Comment:
This doesn't actually fix the testcase for me. I think the parameter type for `max` is inferred to be unsigned. I fixed this up by writing `max<off64_t>` explicitly.